R/log_formatter.R

```diff
@@ -0,0 +1,34 @@
+#' Teal `log_formatter`
+#'
+#' Custom `log_formatter` supporting atomic vectors. By default `glue::glue`
+#' returns n-element vector when vector is provided as an input. This function
+#' generates `"[elem1, elem2, ...]"` for atomic vectors. Function also handles
+#' `NULL` value which normally causes `logger` to return empty character.
+#' @name teal_logger_formatter
+#' @return (`character(1)`) formatted log entry
+#' @keywords internal
+teal_logger_formatter <- function() {
+  logger::log_formatter(
+    function(..., .logcall = sys.call(), .topcall = sys.call(-1), .topenv = parent.frame()) {
+      logger::formatter_glue(
+        ...,
+        .null = "NULL", .logcall = .logcall, .topcall = .topcall, .topenv = .topenv,
```

Comment:
I tested this and the if against `NULL` inside `teal_logger_transformer()` takes precedense and this `.null` argument is redundant

Reply:
Yup. Redundant and also can be handled by `deparse(dput(NULL))`